container/common/fsHandler.go

```diff
@@ -119,16 +149,42 @@ func (fh *realFsHandler) trackUsage() {
 	}
 }
 
+func (fh *realFsHandler) trackInodeUsage() {
+	fh.updateInodeUsage()
+	for {
+		select {
+		case <-fh.stopChan:
+			return
+		case <-time.After(fh.findPeriod):
+			start := time.Now()
+			if err := fh.updateInodeUsage(); err != nil {
+				glog.Errorf("failed to collect filesystem stats using Find - %v", err)
+				fh.findPeriod = fh.findPeriod * 2
+				if fh.findPeriod > maxFindBackoffFactor*fh.minFindPeriod {
+					fh.findPeriod = maxFindBackoffFactor * fh.minFindPeriod
+				}
+			} else {
+				fh.findPeriod = fh.minFindPeriod
+			}
+			duration := time.Since(start)
+			if duration > longFind {
+				glog.V(2).Infof("`find` on following dirs took %v: %v", duration, []string{fh.rootfs, fh.extraDir})
+			}
+		}
+	}
+}
+
 func (fh *realFsHandler) Start() {
-	go fh.trackUsage()
+	go fh.trackDiskUsage()
+	go fh.trackInodeUsage()
```

Comment:
See the note I made in the Issue.  I was able to test this, and the find command works correctly (and takes almost no time, since it always finds 0 inodes) with devicemapper.